incubator/hnc/internal/kubectl/describe.go

```diff
@@ -67,34 +74,72 @@ var describeCmd = &cobra.Command{
 			fmt.Printf("  No children\n")
 		}
 
-		// Early exit if no conditions
+		// Conditions
 		if len(hier.Status.Conditions) == 0 {
 			fmt.Printf("  No conditions\n")
-			return
-		}
-
-		// Conditions
-		fmt.Printf("  Conditions:\n")
-		for _, c := range hier.Status.Conditions {
-			fmt.Printf("  - %s: %s\n", c.Code, c.Msg)
-			if len(c.Affects) == 0 {
-				continue
-			}
-			fmt.Printf("    Affected by this condition:\n")
-			for _, a := range c.Affects {
-				if a.Name != "" {
-					if a.Group == "" {
-						a.Group = "core"
+		} else {
+			fmt.Printf("  Conditions:\n")
+			for _, c := range hier.Status.Conditions {
+				fmt.Printf("  - %s: %s\n", c.Code, c.Msg)
+				if len(c.Affects) == 0 {
+					continue
+				}
+				fmt.Printf("    Affected by this condition:\n")
+				for _, a := range c.Affects {
+					if a.Name != "" {
+						if a.Group == "" {
+							a.Group = "core"
+						}
+						fmt.Printf("      - %s/%s (%s/%s/%s)\n", a.Namespace, a.Name, a.Group, a.Version, a.Kind)
+					} else {
+						fmt.Printf("      - %s\n", a.Namespace)
 					}
-					fmt.Printf("      - %s/%s (%s/%s/%s)\n", a.Namespace, a.Name, a.Group, a.Version, a.Kind)
-				} else {
-					fmt.Printf("      - %s\n", a.Namespace)
 				}
 			}
 		}
+
+		// Events
+		describeEvents(nnm)
 	},
 }
 
+func describeEvents(nnm string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	events, _ := k8sClient.CoreV1().Events(nnm).List(ctx, metav1.ListOptions{})
+	// filter out HNC events only
+	hncEvents := []v1.Event{}
+	for _, event := range events.Items {
+		if event.Source.Component == "hnc.x-k8s.io" {
+			hncEvents = append(hncEvents, event)
+		}
+	}
+	if len(hncEvents) == 0 {
+		fmt.Printf("\nNo recent HNC events for objects in this namespace\n")
+	} else {
+		fmt.Printf("\nEvents from the objects in namespace %s\n", nnm)
+		w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+		fmt.Fprintln(w, "Last Seen\tReason\tObject\tMessage")
+		set := make(map[string]bool)
+		// iterate in reverse order so that we only show the latest event if there's an duplicate
```

Comment:
Are we sure that we'll always get the events in chronological order? Is this documented somewhere? If so, let's include a link. If not, we can _probably_ leave this as-is but we should specify that this is just the behaviour we've _observed_ and we're not sure if it's always true.

Or maybe just re-sort the slice in descending chronological order? See the [`sort` package in the Golang standard library](https://golang.org/pkg/sort/).

Reply:
Used sort as this is easier and always confirm that we have the right order